pkg/internal/util/webhook/webhook.go

```diff
@@ -36,6 +37,14 @@ func IsDNS1123Label(s string) bool {
 	return dns1123LabelRegex.MatchString(s)
 }
 
+type LogLevelSetter interface {
+	SetLogLevel(int)
+}
+
+func SetDefaultLogLevel(spec LogLevelSetter) {
+	spec.SetLogLevel(viper.GetInt("verbose"))
```

Comment:
I'm heavily opposed fetching global variables this deep down the stack.

Reply:
it's public method, so it's not deep, it depends where are you call it and we are calling it in a webhooks, which is also not deep :)

Reply:
It's public method within package, within package, within package. How big is the stack trace where this method gets called? 

What this does is the following:

* Tightly couples viper to the cmd implementation
* We need to register `verbose` flag
* We perform flag validation to `int` only when the webhook is called. (thus you could commit `GetString` and it would fail when the webhook is called, as opposed when program starts. Talk about failing fast.)
* It's difficult injecting test behavior for testing those functions under unit test (if we ever wrote some) since we depend on the global setting

Overall all the reasons why I dislike this approach. Have it as a field in the webhook struct (valid in other circumstances, not what I want from this PR), or fetch is from KubeCarrier object (cached) every time and set it to that value.

Reply:
done